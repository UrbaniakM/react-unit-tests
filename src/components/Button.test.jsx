import * as React from "react";
import { Button } from "./Button";
import { fireEvent, render } from "@testing-library/react";

describe("Button", () => {
  it("renders successfully", () => {
    const buttonLabel = "Button label";

    const { container, getByRole, getByText } = render(<Button>{buttonLabel}</Button>);

    expect(container).toBeDefined();

    expect(getByRole("button")).toBeDefined();
    expect(getByText(buttonLabel)).toBeDefined();
  });

  it("matches snapshot", () => {
    const { container } = render(<Button>Button label</Button>);

    expect(container).toMatchSnapshot();
  });

  it("on click calls onClick function passed as argument", () => {
    const onClickSpy = jest.fn();
    const { getByRole } = render(<Button onClick={onClickSpy} />);

    const button = getByRole("button")
    expect(button).toBeDefined();

    expect(onClickSpy).not.toBeCalled();

    fireEvent.click(button);

    expect(onClickSpy).toBeCalled();
  })
});
