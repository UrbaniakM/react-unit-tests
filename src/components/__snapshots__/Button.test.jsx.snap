// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<div>
  <button>
    Button label
  </button>
</div>
`;
